### ISSUE TITLE
performance degradation using mapZIO function in ZIO Streams

### ISSUE THREAD

**Issue by geazi-anc** [zio/zio#9179]
Hello, everyone! I have been using ZIO Stream to learn more about this library, and I have found an issue out:

My code is pretty simple. It just creates a stream with 1 billion elements and doubles the value of each element. However, when I use `mapZIO` instead of `map`, the performance decreases significantly (from just 9 seconds using `map` to 471 seconds using `mapZIO`).

Is this behavior spected? The codes are below:

Using map to double elements of a stream:

```scala
//> using dep dev.zio::zio:2.1.9
//> using dep dev.zio::zio-streams:2.1.9

import zio.*
import zio.stream.*

object Main extends ZIOAppDefault:
  val app = ZStream.fromIterable(1 to 1000000000).map(_ * 2)
  val run = app.run(ZSink.timed).forEachZIO(e => Console.printLine(e.getSeconds))
// output
//Compiling project (Scala 3.5.0, JVM (21))
// Compiled project (Scala 3.5.0, JVM (21))
// 9
```

Using mapZIO to double each element of a stream:

```scala
//> using dep dev.zio::zio:2.1.9
//> using dep dev.zio::zio-streams:2.1.9

import zio.*
import zio.stream.*

object Main extends ZIOAppDefault:
  val app = ZStream.fromIterable(1 to 1000000000).mapZIO(i => ZIO.succeed(i * 2))
  val run = app.run(ZSink.timed).forEachZIO(e => Console.printLine(e.getSeconds))
// output
// Compiling project (Scala 3.5.0, JVM (21))
// Compiled project (Scala 3.5.0, JVM (21))
// 471
```

**Comment by ghostdogpr** [comment]
See https://blog.pierre-ricadat.com/supporting-high-performance-response-streaming-in-shardcake#heading-chunking-express

tl;dr: `map` is applied on a whole `Chunk` at once in a single operation while `mapZIO` runs an effect for each individual item.

**Comment by geazi-anc** [comment]
Thank you so much for your explanation, @ghostdogpr!

This explanation is there in official documentation of ZIO Stream? It's very useful for all users. If not, maybe add it will be interesting.
